### ISSUE TITLE
pip-compile generates unexpected requirement format (PEP 508 support)

### ISSUE THREAD

**Issue by luzfcb** [jazzband/pip-tools#1418]
https://www.python.org/dev/peps/pep-0508/

pip-compile removes the first part before the @.

#### Environment Versions

1. OS Type: Linux
1. Python version: python3.9
1. pip version: pip 21.1.2
1. pip-tools version: 6.1.0

#### Steps to replicate

1. Create a `base.in` file and include a requirement that uses the PEP508 format

```bash
django-taggit@https://github.com/jazzband/django-taggit/archive/74460fdc4a8b8bf2d36fb8a59691403cde7b4528.zip#sha1=594b7dd32bec37d8bf70a6ffa8866d30e93f3c42
django-debug-toolbar@https://github.com/jazzband/django-debug-toolbar/archive/6ec6bfe0ba5912d7fd0a2b98176a215847157696.zip
```

2. 
```bash
pip-compile -v --output-file base.txt base.in
```

#### Expected result base.txt

```bash
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file=base.txt base.in
#
asgiref==3.3.4
    # via django
django-debug-toolbar@https://github.com/jazzband/django-debug-toolbar/archive/6ec6bfe0ba5912d7fd0a2b98176a215847157696.zip
    # via -r base.in
django-taggit@https://github.com/jazzband/django-taggit/archive/74460fdc4a8b8bf2d36fb8a59691403cde7b4528.zip#sha1=594b7dd32bec37d8bf70a6ffa8866d30e93f3c42
    # via -r base.in
django==3.2.4
    # via
    #   django-debug-toolbar
    #   django-taggit
pytz==2021.1
    # via django
sqlparse==0.4.1
    # via
    #   django
    #   django-debug-toolbar

```

#### Actual result base.txt

```bash
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file=base.txt base.in
#
asgiref==3.3.4
    # via django
https://github.com/jazzband/django-debug-toolbar/archive/6ec6bfe0ba5912d7fd0a2b98176a215847157696.zip
    # via -r base.in
https://github.com/jazzband/django-taggit/archive/74460fdc4a8b8bf2d36fb8a59691403cde7b4528.zip#sha1=594b7dd32bec37d8bf70a6ffa8866d30e93f3c42
    # via -r base.in
django==3.2.4
    # via
    #   django-debug-toolbar
    #   django-taggit
pytz==2021.1
    # via django
sqlparse==0.4.1
    # via
    #   django
    #   django-debug-toolbar

```

**Comment by atugushev** [comment]
Most likely this is going to be fixed in #1392.

**Comment by AndydeCleyre** [comment]
Yup, this has been fixed by the mentioned PR, and will be in the next release!
